tests/test_bids_file2.m

```diff
@@ -70,11 +72,175 @@ function test_change()
   file = file.set_entity('task', 'faceRecognition');
   file = file.set_entity('run', '01');
   assertEqual(file.filename, 'sub-02_task-faceRecognition_acq-abc_run-01_test.a');
+
+end
+
+function test_bids_file_remove_entities()
+
+  % GIVEN
+  filename = 'sub-01_ses-test_task-faceRecognition_run-02_bold.nii';
+  file = bids.File2(filename, 'use_schema', false);
+  % WHEN
+  file.entities.run = '';
+  % THEN
+  assertEqual(file.filename, 'sub-01_ses-test_task-faceRecognition_bold.nii');
+
 end
 
 function test_reorder()
+
   filename = 'wuasub-01_task-faceRecognition_ses-test_run-02_bold.nii';
   file = bids.File2(filename, 'use_schema', false);
   file = file.reorder_entities({'sub', 'ses'});
+
+  assertEqual(file.entity_order, {'sub'; 'ses'; 'task'; 'run'});
   assertEqual(file.json_filename, 'wuasub-01_ses-test_task-faceRecognition_run-02_bold.json');
+
+end
+
+function test_reorder_by_property()
+
+  % This fails but not sure we want to do things that way anyway
```

Comment:
Is this a "API" we want to support? Maybe better to only have one way of doing things, no?